src/helpers/map.js

```diff
@@ -173,3 +177,100 @@ export function findApproxTile(targetTile) {
   console.error(`Unable to find tile: ${targetTile}`);
   return {};
 }
+
+
+/*
+ * Initializes and returns a 2D array with the specified width, height, and
+ * default value.
+ *
+ * width: the width of the initialized 2D array
+ * height: the height of the initialized 2D array
+ * defaultVal: the value to give each element in the initialized 2D array
+ */
+export function init2dArray(width, height, defaultVal) {
+  const matrix = [];
+
+  for (let x = 0; x < width; x++) {
+    matrix[x] = new Array(height);
+    for (let y = 0; y < height; y++) {
+      matrix[x][y] = defaultVal;
+    }
+  }
+
+  return matrix;
+}
+
+
+/*
+ * Returns the sum of all corresponding elements from two matrices being
+ * multiplied together. For example:
+ *   a = 1 2   b = 4 1
+ *       3 4       2 3
+ *   multiplyCorrespondingElementsAndSum(a, b)
+ *   (1*4) + (2*1) + (3*2) + (4*3) = 21
+ *
+ * m1: the first matrix
+ * m2: the second matrix (of the same dimensions as m1)
+ */
+export function multiplyCorrespondingElementsAndSum(m1, m2) {
+  const mWidth = m1.length;
+  const mHeight = m1[0].length;
+  let sum = 0;
+
+  for (let x = 0; x < mWidth; x++) {
+    for (let y = 0; y < mHeight; y++) {
+      sum += m1[x][y] * m2[x][y];
+    }
+  }
+
+  return sum;
+}
+
+
+/* Returns a matrix with a buffer applied around its perimeter. Example:
+ *   Take the element values of the matrix and apply a bufSize perimeter
+ *   of value bufVal.
+ *     mWithBuf = 1 2 3   bufSize = 2   mWithBuf = 1 1 1 1 1 1 1
+ *                4 5 6                            1 1 1 1 1 1 1
+ *                7 8 9   bufVal = 1               1 1 1 2 3 1 1
+ *                                                 1 1 4 5 6 1 1
+ *                                                 1 1 7 8 9 1 1
+ *                                                 1 1 1 1 1 1 1
+ *                                                 1 1 1 1 1 1 1
+ *
+ * m: a 2D array, which will have a buffer added around its perimeter
+ * bufSize: the size of buffer to add around the matrix
+ * bufVal: the value to fill the buffer with
+ */
+export function addBufferTo2dArray(m, bufSize, bufVal) {
+  const mWidth = m.length;
+  const mHeight = m[0].length;
+  const mWithBufWidth = mWidth + (bufSize * 2);
+  const mWithBufHeight = mHeight + (bufSize * 2);
+
+  // Create a grid with matrix's values and a bufSize perimeter of zeros
+  const mWithBuf = init2dArray(mWithBufWidth, mWithBufHeight, bufVal);
+  fillGridWithSubgrid(mWithBuf, m, bufSize, bufSize);
+
+  return mWithBuf;
+}
+
+
+export function getSubarrayFrom2dArray(array, xCenter, yCenter, width, height) {
```

Comment:
Ya you could just put them at the top. I believe code will error with `AssertionError` or something like that. The idea is, if someone gives an input they're not supposed to, rather than pass them back something that's not what they expect, throw an error so they know they made a mistake.